app/models/manageiq/providers/container_manager.rb

```diff
@@ -67,16 +67,12 @@ def supports_metrics?
       true
     end
 
-    # required by aggregate_hardware
-    alias :all_computer_systems :computer_systems
-    alias :all_computer_system_ids :computer_system_ids
```

Comment:
you know, I think you could use `host_hardwares` and most of these would work...

```ruby
has_many :host_hardwares, :through => :computer_systems, :as => :hardware, :class_name => "Hardware"
```

(only if you feel it will be less work to use the already defined relations)
not sure if you'd need to change the mixin to only add the methods if a relation was already defined

Reply:
i like this but, sort of unrelated, i still think we've an issue with sum: 
```
2.5.5 :009 > ManageIQ::Providers::ContainerManager.first.aggregate_cpu_total_cores
...  SELECT SUM("hardwares"."cpu_total_cores") FROM "hardwares" WHERE "hardwares"."computer_system_id" IN (SELECT "computer_systems"."id" FROM "computer_systems" INNER JOIN "container_nodes" ON "computer_systems"."managed_entity_id" = "container_nodes"."id" WHERE "container_nodes"."ems_id" = $1 AND "computer_systems"."managed_entity_type" = $2 AND "container_nodes"."deleted_on" IS NULL)  [["ems_id", 41], ["managed_entity_type", "ContainerNode"]]
 => 6
2.5.5 :010 > ManageIQ::Providers::ContainerManager.sum(:aggregate_cpu_total_cores)
   (0.9ms)  SELECT SUM((SELECT SUM("hardwares"."cpu_total_cores") FROM "hosts" INNER JOIN "hardwares" ON "hardwares"."host_id" = "hosts"."id" WHERE "hosts"."ems_id" = "ext_management_systems"."id")) FROM "ext_management_systems" WHERE "ext_management_systems"."type" IN ('ManageIQ::Providers::ContainerManager', 'ManageIQ::Providers::Kubernetes::ContainerManager', 'ManageIQ::Providers::Openshift::ContainerManager')
 => 0.0
```

Reply:
no sorry container manager just is interesting